modules/controller/index.js

```diff
@@ -1,138 +1,127 @@
-
-  return function(request, response, next) {
-    var err = validateRequest(request, response, source);
+Controller.prototype.read = function (opts) {
+  var config = this._configureController('read', opts);
+  return requestHandler(config);
+};
 
-    if (err) {
-      return responder(polyResponse(err), request, response);
-    }
+Controller.prototype.readRelation = function (opts) {
+  var config = this._configureController('readRelation', opts);
+  return requestHandler(config);
+};
 
-    handle(opts, request, response, next);
-  };
+Controller.prototype.update = function (opts) {
+  var config = this._configureController('update', opts);
+  return requestHandler(config);
 };
 
-Controller.prototype._lookupValidateRequest = function(method) {
-  switch (method) {
-    case 'readRelation':
-      return relationExistsReqValidator;
-    case 'create':
-    case 'update':
-      return contentTypeReqValidator;
-    default:
-      return function() {};
-  }
+Controller.prototype.destroy = function (opts) {
+  var config = this._configureController('destroy', opts);
+  return requestHandler(config);
 };
 
-Controller.prototype._lookupPolyResponse = function(method) {
-  switch (method) {
-    case 'read':
-    case 'readRelation':
-      return readResponse;
-    case 'create':
-      return createResponse;
-    case 'update':
-      return updateResponse;
-    case 'destroy':
-      return destroyResponse;
-    case 'lookupFailed':
-      return lookupFailedResponse;
-    default:
-      return readResponse;
-  }
+Controller.prototype._create = function(opts, request) {
+  return this.source.create(opts.method, request.body.data);
 };
 
-Controller.prototype._lookupHandler = function(method) {
-  switch (method) {
-    case 'create':
-      return this._createHandler;
-    case 'read':
-      return this._readHandler;
-    case 'readRelation':
-      return this._readRelatedHandler;
-    case 'update':
-      return this._updateHandler;
-    case 'destroy':
-      return this._destroyHandler;
-    default:
-      return this._readHandler;
-  }
+Controller.prototype._read = function(opts, request) {
+  var includes = this._includes(request);
+  var filters = this._filters(request);
+  return this.source.read({
+    relations: includes.length ? includes : opts.include,
+    filters: Object.keys(filters).length ? filters : opts.filters
+  });
 };
 
-Controller.prototype._createHandler = function(opts, request, response, next) {
-  var dataFromSource = this._dataFromSource.bind(this);
-  dataFromSource(request.body.data, opts, createResponse, request, response, next);
+Controller.prototype._readRelation = function(opts, request) {
+  var source = this.source;
+  var relation = request.params ? request.params.relation : null;
+  return source.byId(request.params.id, relation).then(function (model) {
+    if (!model) {
+      // this isn't going to cause a 404 correctly
+      throw new Error('Unable to locate model.');
+    }
```

Comment:
This gets picked up in the catch on line 22 of the request handler here: https://github.com/endpoints/endpoints/pull/39/files#diff-9a4c3ed2ac4f045b8d71d4bfe7cd5b11R22

You could do `var err = new Error('Unable to locate model.'); err.httpStatus = 404; err.title = 'File not Found';`

and make the read payload lib pass those properties through like so: https://github.com/endpoints/endpoints/blob/pr-for-bob/modules/controller/lib/payloads/update.js#L10-L15